src/braket/tasks/gate_model_quantum_task_result.py

```diff
@@ -31,6 +32,10 @@
 T = TypeVar("T")
 
 
+def result_type_hash(rt_type):
+    return repr(dict(sorted(dict(rt_type).items(), key=lambda x: x[0])))
```

Comment:
Can you add unit tests asserting that result types are hashed properly, particularly arbitrary Hermitian observables? Also, this can be private, and moved to the bottom of the class.

Reply:
On it